### PR TITLE
Fix(Sales): Resolve "characters not found" issue

### PR DESCRIPTION
**Pull request by SpeedyD**
Referring to Issue #459 that is-

I feel like this is a form of band-aid solution, but it should look and work decently. Please look it over thoroughly.

### CODE REVIEW

File: resources/views/admin/sales/_character_select_entry.blade.php [hunk 1]
```diff
@@ -14,7 +14,7 @@
                 <a href="#" class="float-right fas fa-close"></a>
                 <div class="form-group">
                     {!! Form::label('slug[]', 'Character Code') !!}
-                    {!! Form::text('slug[]', $character->character->slug, ['class' => 'form-control character-code']) !!}
+                    {!! Form::text('slug[]', $character->character ? $character->character->slug : null, ['class' => 'form-control character-code']) !!}
```
Comment (AW0005) [inline resources/views/admin/sales/_character_select_entry.blade.php:17]: Nit: I think you should be able to use the `??` operator here to make these a little shorter like this:
`$character->character->slug ?? null`

**Review by AW0005 — APPROVED** [review]
I left a little nitpick comment and thought the rest of this looked good!

But I also noticed in the description you felt like this was a band-aid solution so I did a little poking around and I think another option might be to update the relationship on `Sales` for the `SalesCharacter to be this:
`$this->hasMany('App\Models\Sales\SalesCharacter', 'sales_id')->has('character')`

So it would never return SalesCharacters as a part of the collection at all, if their character is undefined. The downside to that though, is then there's no messaging to the user that a character was deleted and that's why it's missing. 🤔 Not sure which would be more preferable.

**Comment by itinerare** [comment]
Yeah I had the idea to have better checking on the interface itself so that the information entered pertinent to the sale was still accessible; ideally I think this would be the case for both the admin and public-facing interfaces so that records are maintained, just replacing relevant bits with [deleted character] or somesuch as is done for e.g. logs.

**Comment by SpeedyD** [comment]
@AW0005, @itinerare had personally told me they'd look things over and work on it, so I'm leaving it 100% up to them from here. 
I just wanted to set up a temporary solution that worked.. and it does, even if it's not that clean. ^^;

File: resources/views/admin/sales/_character_select_entry.blade.php (hunk 1 above)
Comment (SpeedyD) [reply resources/views/admin/sales/_character_select_entry.blade.php:17]: Would you believe I wasn't aware of ternary operators 3 months ago? Anyway, good note for @itinerare if/when they fix this. ^^;

**Comment by itinerare** [comment]
Yeah I had some bold thoughts about getting character info by virtue of it being soft-deleted rather than deleted-deleted... though tbh this probably wouldn't work as well as I'd like considering that iirc the core character is the only thing to which that applies and not e.g. the image, traits, etc, which are kinda the relevant bits of info that would contextualize the sale info. At this rate probably what I would end up doing is just cleaning up the display in line w my comment above.

**Comment by SpeedyD** [comment]
Just refreshing the branch by putting all the new stuff in..

**Comment by SpeedyD** [comment]
..Hrm, I have an idea how to do this the way Merc maybe wants, but I think I need #615 resolved first. It might help the process along..

So I'm converting this to draft for now.

**Comment by SpeedyD** [comment]
Don't know if still needed, but I've opted to at least push the dev branch in for now, resolving the one conflict

**Comment by SpeedyD** [comment]
..Scratch that. Looking at the changes that have been done and the two edits in here do not even get reached anymore. Closing PR.

**Pull request closed without merging**